db/active_replicator_checkpointer.go

```diff
@@ -78,16 +80,22 @@ type CheckpointerStats struct {
 
 func NewCheckpointer(ctx context.Context, clientID string, configHash string, blipSender *blip.Sender, replicatorConfig *ActiveReplicatorConfig, statusCallback statusFunc) *Checkpointer {
 	return &Checkpointer{
-		clientID:                       clientID,
-		configHash:                     configHash,
-		blipSender:                     blipSender,
-		activeDB:                       replicatorConfig.ActiveDB,
-		expectedSeqs:                   make([]SequenceID, 0),
-		processedSeqs:                  make(map[SequenceID]struct{}),
-		idAndRevLookup:                 make(map[IDAndRev]SequenceID),
-		checkpointInterval:             replicatorConfig.CheckpointInterval,
-		ctx:                            ctx,
-		stats:                          CheckpointerStats{},
+		clientID:           clientID,
+		configHash:         configHash,
+		blipSender:         blipSender,
+		activeDB:           replicatorConfig.ActiveDB,
+		expectedSeqs:       make([]SequenceID, 0),
+		processedSeqs:      make(map[SequenceID]struct{}),
+		idAndRevLookup:     make(map[IDAndRev]SequenceID),
+		checkpointInterval: replicatorConfig.CheckpointInterval,
+		ctx:                ctx,
+		stats: CheckpointerStats{
+			ProcessedSequenceLen:            *replicatorConfig.ReplicationStatsMap.ProcessedSequenceLen,
```

Comment:
Dereferencing these here means the Checkpointer stats will be a copy of the original stat - any increments done on c.stats will not be reflected in the ReplicationStatsMap instances.